### PR TITLE
fix: Persons Search backend

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

First part of https://github.com/PostHog/posthog/pull/11549

## Changes

* Backend changes only adding new API endpoint

### CODE REVIEW

File: ee/clickhouse/views/person.py [hunk 1]
```diff
@@ -17,31 +17,10 @@ def funnel_correlation(self, request: request.Request, **kwargs) -> response.Res
...
-            data={
-                "results": [{"people": people, "count": len(people)}],
-                "next": next_url,
-                "initial": initial_url,
-                "is_cached": results_package.get("is_cached"),
-                "last_refresh": results_package.get("last_refresh"),
-                "missing_persons": missing_persons,
-            }
-        )
+        return _respond_with_cached_results(self.calculate_funnel_correlation_persons(request))
```
Comment (neilkakkar) [inline ee/clickhouse/views/person.py:20]: Changing how the cached result is used means that all existing insights will start failing for upto 7 days 😬 . (oh, the troubles with cache invalidation).

You could clear the cache, or keep the cached result the same, and add the mods to `_respond_with_cached_results`, vs to `calculate_funnel_*`

Comment (benjackwhite) [reply ee/clickhouse/views/person.py:20]: Yeah I just noticed this as well. Annoying. Fix coming up

File: posthog/api/action.py [hunk 2]
```diff
@@ -184,10 +184,15 @@ def list(self, request: request.Request, *args: Any, **kwargs: Any) -> Response:
         actions_list: List[Dict[Any, Any]] = self.serializer_class(actions, many=True, context={"request": request}).data  # type: ignore
         return Response({"results": actions_list})
 
+    # NOTE: Deprecated in favour of `persons/trends` endpoint
+    # Once the old way of exporting CSVs is removed, this endpoint can be removed
     @action(methods=["GET"], detail=False)
     def people(self, request: request.Request, *args: Any, **kwargs: Any) -> Response:
         team = self.team
         filter = Filter(request=request, team=self.team)
+        if not filter.limit:
```
Comment (neilkakkar) [inline posthog/api/action.py:193]: this seems wrong, my fix for trends for person-on-events that went in a day or two ago should've gotten rid of the `LIMIT 200`

(apologies about the merge conflict)

File: posthog/queries/trends/test/__snapshots__/test_person.ambr [hunk 3]
```diff
@@ -18,7 +18,7 @@
        AND timestamp >= toDateTime('2021-01-21 00:00:00')
        AND timestamp <= toDateTime('2021-01-21 23:59:59') )
   GROUP BY actor_id
-  LIMIT 100
+  LIMIT 200
```
Comment (neilkakkar) [inline posthog/queries/trends/test/__snapshots__/test_person.ambr:21]: needs a revert

File: posthog/queries/trends/trends_actors.py [hunk 4]
```diff
@@ -138,7 +137,7 @@ def actor_query(self, limit_actors: Optional[bool] = True) -> Tuple[str, Dict]:
                 limit="LIMIT %(limit)s" if limit_actors else "",
                 offset="OFFSET %(offset)s" if limit_actors else "",
             ),
-            {**params, "offset": self._filter.offset, "limit": limit},
+            {**params, "offset": self._filter.offset, "limit": self._filter.limit or 200},
```
Comment (neilkakkar) [inline posthog/queries/trends/trends_actors.py:140]:
```suggestion
            {**params, "offset": self._filter.offset, "limit": self._filter.limit or 100},
```

Comment (benjackwhite) [reply posthog/queries/trends/trends_actors.py]: Done

File: posthog/api/action.py (hunk 2 above)
Comment (neilkakkar) [reply posthog/api/action.py:193]: My goal here: It should be 100, not 200 😅

Comment (benjackwhite) [reply posthog/api/action.py:193]: Gotcha 👍

**Review by neilkakkar — APPROVED** [review]

**Comment by benjackwhite** [comment]
@neilkakkar special thanks for your efforts on discussion and reviewing all of this. Learnt a tonne in the process about bits of PostHog I never touched 🙌

**Comment by neilkakkar** [comment]
Nicely done, thanks for sticking around through all the iterations 😅

**Pull request merged**